pgcli/pgexecute.py

```diff
@@ -190,6 +184,16 @@ def connect(self, database=None, user=None, password=None, host=None,
             self.conn.close()
         self.conn = conn
         self.conn.autocommit = True
+
+        if dsn:
+            # When we connect using a DSN, we don't really know what db,
+            # user, etc. we connected to. Let's read it.
+            # Note: moved this after setting autocommit because of #664.
+            db = self._select_one(cursor, 'select current_database()')[0]
+            user = self._select_one(cursor, 'select current_user')[0]
+            host = self._select_one(cursor, 'select inet_server_addr()')[0]
+            port = self._select_one(cursor, 'select inet_server_port()')[0]
+
```

Comment:
Can we collapse this into a single sql call instead of 4 calls? 

```
select current_database(), currect_user, inet_server_addr, inet_server_port
```

and assign the results to the appropriate variables? This should cut down on the start up time if the database instance is remote and there is a noticeable latency in the connection.